File: packages/typescript/tests/system/frames.test.ts
import { describe } from "vitest";
import { baseIt } from "./helpers.js";

describe("Frames", () => {
  const it = baseIt.override("setup", async ({ setup, skip }) => {
    return async (options) => {
      const result = await setup(options);
      const { driverType } = result;

      // Frames support is not implemented in Appium yet
      if (driverType === "appium")
        skip("Frames support is not implemented in Appium yet");

      return result;
    };
  });

  it("nested frames", async ({ expect, setup }) => {
    const { al, $ } = await setup();

    await $.navigate("https://the-internet.herokuapp.com/nested_frames");

    await al.do("click MIDDLE text");

    const texts = await al.get("text from all frames");
    expect(texts).toEqual(["LEFT", "MIDDLE", "RIGHT", "BOTTOM"]);
  });

  it("cross origin iframe", async ({ expect, setup }) => {
    const { al, $ } = await setup();

    await $.navigate("cross_origin_iframe.html");

    await al.check("button 'Main Page Button' is present", {
      assert: expect.assert,
    });
    await al.do("click button 'Click Me Inside Iframe'");
    await al.check("text 'Button Clicked!' is present", {
      assert: expect.assert,
    });
    await al.do("click link 'Iframe Link'");
    await al.check("text 'Link Clicked!' is present", {
      assert: expect.assert,
    });
  });
});

File: packages/java/src/test/java/ai/alumnium/system/FramesTest.java
package ai.alumnium.system;

import java.io.File;
import java.util.List;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.condition.DisabledIfEnvironmentVariable;

@DisabledIfEnvironmentVariable(named = "ALUMNIUM_DRIVER", matches = "appium.*")
public class FramesTest extends BaseTest {

  private static final String CROSS_ORIGIN_IFRAME_URL =
      new File("../python/examples/support/pages/cross_origin_iframe.html").toURI().toString();

  @Test
  void testNestedFrames() {
    navigate("https://the-internet.herokuapp.com/nested_frames");

    al.act("click MIDDLE text");
    Assertions.assertEquals(
        List.of("LEFT", "MIDDLE", "RIGHT", "BOTTOM"), al.get("text from all frames"));
  }

  @Test
  void testCrossOriginIframe() {
    navigate(CROSS_ORIGIN_IFRAME_URL);

    al.check("button 'Main Page Button' is present");
    al.act("click button 'Click Me Inside Iframe'");
    al.check("text 'Button Clicked!' is present");
    al.act("click link 'Iframe Link'");
    al.check("text 'Link Clicked!' is present");
  }
}
